syntax = "proto3";

package Fq.Private.V1;

import "ydb/core/yq/libs/protos/fq_private.proto";

service FqPrivateTaskService {
    // gets new task
    rpc GetTask(Fq.Private.GetTaskRequest) returns (Fq.Private.GetTaskResponse);

    // pings new task (also can update metadata)
    rpc PingTask(Fq.Private.PingTaskRequest) returns (Fq.Private.PingTaskResponse);

    // writes rows
    rpc WriteTaskResult(Fq.Private.WriteTaskResultRequest) returns (Fq.Private.WriteTaskResultResponse);

    //Nodes
    rpc NodesHealthCheck(Fq.Private.NodesHealthCheckRequest) returns (Fq.Private.NodesHealthCheckResponse);
}
